internal/worker/bootstrap/worker.go

```diff
@@ -250,6 +290,51 @@ func (w *bootstrapWorker) bootstrapParams(ctx context.Context, dataDir string) (
 	return args, nil
 }
 
+// getEnviron creates a new environ using the provided NewEnvironFunc from the
+// worker config.
+func (w *bootstrapWorker) getEnviron(ctx context.Context) (environs.Environ, error) {
+	controllerModel, err := w.cfg.SystemState.Model()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var cred *cloud.Credential
+	if cloudCredentialTag, ok := controllerModel.CloudCredentialTag(); ok {
+		credentialValue, err := w.cfg.CredentialService.CloudCredential(ctx, credential.IdFromTag(cloudCredentialTag))
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		cloudCredential := cloud.NewNamedCredential(
+			credentialValue.Label,
+			credentialValue.AuthType(),
+			credentialValue.Attributes(),
+			credentialValue.Revoked,
+		)
+		cred = &cloudCredential
+	}
```

Comment:
Also can we put that into another method, just to make readability nicer. `getEnvironCredential`.

Reply:
done